src/lib/hooks/useStaking.tsx

```diff
@@ -0,0 +1,210 @@
+import { useMemo, useCallback } from 'react';
```

Comment:
Leave this for now as we can do this in it's own PR - We should extract the error text in the file to a global error constant file and replace with all instances in the codebase, I think we re-use them quite a bit.